### DISCUSSION TITLE
[Bug] A question called "sw8-correlation" Header is null

### DISCUSSION THREAD

**Discussion by zhifu-guanxi** [apache/skywalking#8149] [Q&A]
### Search before asking

- [X] I had searched in the [issues](https://github.com/apache/skywalking/issues?q=is%3Aissue) and found no similar issues.

### Apache SkyWalking Component

Java Agent (apache/skywalking-java)

### What happened

The version I'm using is 8.4.0.
When I call the third-party service with resttemplate after I mount the Javaagent,  it returns HTTP 400.
It is normal for me to use postman to call this interface.
After I debugged the code, I found that there were more headers on the Request after skywalking was mounted，such as "sw8"、"sw8-correlation"、"sw8-x".
The problem is that the header of "sw8-correlation" is an empty string. 
Some versions of springboot applications will return bad request. 
I tried to add content to "sw8-correlation", but failed.
Finally, I can only add code in the "aftermethod" of org.apache.skywalking.apm.plugin.spring.resttemplate.sync.Restrequestinterceptor.class.

### What you expected to happen

Remove “ sw8-correlation” or add content to it.
It would be better if I could understand the design principle of “ sw8-correlation”, I guess it is related to static method calls and I don't know if it is correct.

### How to reproduce

Finally, I can only add code in the "aftermethod" of org.apache.skywalking.apm.plugin.spring.resttemplate.sync.Restrequestinterceptor.class.
`
    public Object afterMethod(EnhancedInstance objInst, Method method, Object[] allArguments, Class<?>[] argumentsTypes,
        Object ret) throws Throwable {
        ClientHttpRequest clientHttpRequest = (ClientHttpRequest) ret;
        if (clientHttpRequest instanceof AbstractClientHttpRequest) {
            AbstractClientHttpRequest httpRequest = (AbstractClientHttpRequest) clientHttpRequest;
            ContextCarrier contextCarrier = RestTemplateRuntimeContextHelper.getContextCarrier();
            CarrierItem next = contextCarrier.items();
            while (next.hasNext()) {
                next = next.next();
                **if (StringUtil.isNotEmpty(next.getHeadValue())) {**
                    httpRequest.getHeaders().set(next.getHeadKey(), next.getHeadValue());
                **}**
            }
        }
        return ret;
    }`

### Are you willing to submit PR?

- [ ] Yes I am willing to submit a PR!

### Code of Conduct

- [X] I agree to follow this project's [Code of Conduct](https://www.apache.org/foundation/policies/conduct)

**Comment by wu-sheng** [comment]
These headers are called tracing context.

You could read this for more details. https://skywalking.apache.org/docs/main/latest/en/protocols/readme/#tracing

**Comment by homeewu** [comment]
hello , i still have this problem . I use skywalking-agent:8.13.0 , when i call outside service , their firewall forbid my request for the reason sw8-correlation header add  (space) value. 
how to sovle this problem , can anyone help me.

**Reply by wu-sheng** [reply]
If you are calling a strict 3rd party implementation, you have 3 options
1. Remove the plugin of the RPC framework you are using.
2. Change the source codes of plugin mentioned in <1> and remove the header injection.
3. Add a gateway before calling 3rd party, remove the three headers out of the HTTP request.

**Reply by homeewu** [reply]
so you mean such as those apm-httpClient-4.x-plugin-8.13.0.jar in plugins folder , i need to remove them, right ?

**Reply by wu-sheng** [reply]
I don't know which one. The one you used sending request to the 3rd party system.
